### PR TITLE
DDF-6075 Dependency Upgrades

### PR DESCRIPTION
**Pull request by josephthweatt**
#### What does this PR do?
Dependency upgrades for:
Upgrades to dependencies in the root pom, including:
xerces
netty/netty-codec

Upgrade to snakeyaml in cxf's features.xml

#### Who is reviewing it? 
@aaronilovici 
@mdang8 

#### How should this be tested?
A full build 

#### What are the relevant tickets?
Fixes: #6075 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by josephthweatt** [comment]
build now

**Comment by josephthweatt** [comment]
build now

File: pom.xml [hunk 1]
```diff
@@ -256,7 +256,8 @@
         <mina.version>2.0.6</mina.version>
         <mpegts-streamer.version>0.1.0_2</mpegts-streamer.version>
         <net.minidev.asm.version>1.0.2</net.minidev.asm.version>
-        <netty.version>4.1.32.Final</netty.version>
+        <netty.version>4.1.44.Final</netty.version>
```
Comment (aaronilovici) [inline pom.xml:259]: Instead of separating these out, can we just upgrade netty as a whole to 46 instead of just netty codec?

File: pom.xml [hunk 2]
```diff
@@ -209,10 +209,10 @@
         <httpmime.version>${httpclient.version}</httpmime.version>
         <jai-imageio-core.version>1.3.1</jai-imageio-core.version>
         <JavaAPIforKml.version>2.2.1_CODICE_1</JavaAPIforKml.version>
-        <jackson.version>2.9.10</jackson.version>
+        <jackson.version>2.10.0</jackson.version>
         <!-- databind 2.9.10.1 fixes CVE-2019-16942, CVE-2019-16943, CVE-2019-17531.
         This separate version can be removed when jackson.version is upgraded to a version newer than 2.9.10. -->
-        <jackson.databind.version>2.9.10.1</jackson.databind.version>
+        <jackson.databind.version>2.10.0</jackson.databind.version>
```
Comment (aaronilovici) [inline pom.xml:215]: If these are matchingnow (jackson and jackson databind) you should remove the comment and refactor all instances of jackson.databind.version -> jackson.version

**Comment by josephthweatt** [comment]
build now

File: features/pom.xml [hunk 3]
```diff
@@ -62,12 +62,12 @@
         <geronimo-jta-spec-version>1.1.1</geronimo-jta-spec-version>
         <camel.guava.version>19.0</camel.guava.version>
         <cxf.jackson.version>${jackson.version}</cxf.jackson.version>
-        <cxf.jackson.databind.version>${jackson.databind.version}</cxf.jackson.databind.version>
+        <cxf.jackson.databind>${jackson.databind}</cxf.jackson.databind>
```
Comment (aaronilovici) [inline features/pom.xml:65]: You can remove this too. All dependencies should use ${jackson.version} or ${cxf.jackson.version}

In fact I don't think you have ${jackson.databind} defined anywhere.

**Comment by josephthweatt** [comment]
build now

File: features/pom.xml (hunk 3 above)
Comment (aaronilovici) [reply features/pom.xml:65]: Yeah that's the reason for the build failure ^

[2020-05-22T14:52:33.469Z] [ERROR] [ERROR] Some problems were encountered while processing the POMs:
[2020-05-22T14:52:33.469Z] [ERROR] 'dependencies.dependency.version' for com.fasterxml.jackson.core:jackson-databind:jar must be a valid version but is '${jackson.databind}'. @ line 102, column 22
[2020-05-22T14:52:33.469Z] [ERROR] 'dependencies.dependency.version' for com.fasterxml.jackson.core:jackson-databind:jar must be a valid version but is '${jackson.databind}'. @ line 373, column 22
[2020-05-22T14:52:33.469Z] [ERROR] 'dependencies.dependency.version' for com.fasterxml.jackson.core:jackson-databind:jar must be a valid version but is '${jackson.databind}'. @ line 52, column 22

**Comment by josephthweatt** [comment]
build now

**Comment by stustison** [comment]
Some of these changes are already in Josh's PR: https://github.com/codice/ddf/pull/6070

**Comment by josephthweatt** [comment]
build now

**Comment by josephthweatt** [comment]
build now

**Comment by josephthweatt** [comment]
build now

**Pull request merged**